src/sagemaker/serve/builder/model_builder.py

```diff
@@ -118,7 +118,8 @@ class ModelBuilder(Triton, DJL, JumpStart, TGI, Transformers):
             into a stream. All translations between the server and the client are handled
             automatically with the specified input and output.
         model (Optional[Union[object, str]): Model object (with ``predict`` method to perform
```

Comment:
is this ID:Task format clear for the customer? What if we need more identifiers does this extend well?  Is parsing a string the best way to pass parameters?

Have we considered a dictionary here?
```
model = {
    "ModelID": "bert-based-uncased",
    "Task": "Fill-Mask",
    "Version": "v1.2.0" <- easily extend to more fields
}
```
This makes the fields explicit

Reply:
We had some discussions on this. I can share with you

Reply:
The gist is that `ModelBuilder` makes the assumption in a lot of places that `self.model` is a string. Internally all the model servers are treating the `self.model` as a string as well. Adding a field to model and treating it as an object will require an overhaul of `ModelBuilder`. Hence the least disruptive way forward, as suggested by @mohanasudhan, is to append`:task` after `model_id`.

Reply:
Would just passing via env_vars be the least disruptive? Its very likely that a customer would already be doing this. According to https://github.com/aws/sagemaker-huggingface-inference-toolkit/issues/112#issuecomment-1953637502 it looks like all huggingface guides recommend passing the HF_TASK explicitly.

Reply:
Can you address @gwang111  comment?

Reply:
Is there a use-case for overriding more HF attributes? 

There is no use-case for overriding MODEL_REVISION today. There is also no such field as "version".  I presume version is used to indicate that an API is changing fields. Versions are hard to maintain in packages and lead to problems with deprecation.

Adding HF_TASK to Model_Builder is another option we have discussed but have received pushback on. I don't think adding HF_TASK to env variables is the same as adding it to the Model attributes. 

There are two high level concerns we seem to want to address here:
1. HF_TASK needs to be model attribute
2. HF_TASK needs to be explicit
3. Ability to extend fields (not sure if there are more fields, but could change).

Low level design options:
A. Add Task to Model - Invasive change with existing implementations of DLC servers - meets 1, 2, 3 (might not be necessary for all servers)
B. Add Task or alternatively HF_METADATA to ModelBuilder  - meets 2, 3 does not meet 1
C. Add Task to Model ID string - meets 1, 2, not 3 
D. Add Task to env var - meets 2, 3 not 1

Reply:
Update: Follow up with HuggingFace team @philschmid if there are additional HF attributes to override and preferred approach

Reply:
Resolving this by introducing a new field `model_metadata` to ModelBuilder, holding all possible HF attributes to be overridden